clouddriver/clouddriver-artifacts/src/test/java/com/netflix/spinnaker/clouddriver/artifacts/s3/S3ArtifactCredentialsTest.java

```diff
@@ -112,88 +110,83 @@ class S3ArtifactCredentialsTest {
+            .region(Region.of(localstack.getRegion()))
+            .credentialsProvider(
+                StaticCredentialsProvider.create(
+                    AwsBasicCredentials.create(
+                        localstack.getAccessKey(), localstack.getSecretKey())))
+            .forcePathStyle(true)
+            .overrideConfiguration(
+                ClientOverrideConfiguration.builder()
+                    .addExecutionInterceptor(
+                        new S3ArtifactCredentials.S3ArtifactRequestInterceptor(
+                            "my-s3-account", new S3ArtifactProviderProperties()))
+                    .addExecutionInterceptor(new SpectatorExecutionInterceptor(registry))
+                    .build())
             .build();
 
     // Create a bucket so there's a place to retrieve from
-    amazonS3.createBucket(BUCKET_NAME);
+    s3Client.createBucket(CreateBucketRequest.builder().bucket(BUCKET_NAME).build());
 
     // Create a file so there's something to download
-    amazonS3.putObject(BUCKET_NAME, KEY_NAME, CONTENTS);
+    s3Client.putObject(
+        PutObjectRequest.builder().bucket(BUCKET_NAME).key(KEY_NAME).build(),
+        RequestBody.fromString(CONTENTS));
   }
 
   @Test
   void normalDownload() throws IOException {
     S3ArtifactCredentials s3ArtifactCredentials =
-        new S3ArtifactCredentials(account, amazonS3, s3ArtifactProviderProperties);
+        new S3ArtifactCredentials(account, s3Client, s3ArtifactProviderProperties, registry);
     try (InputStream artifactStream = s3ArtifactCredentials.download(artifact)) {
       String actual = new String(artifactStream.readAllBytes(), StandardCharsets.UTF_8);
       assertEquals(CONTENTS, actual);
     }
 
     // Verify that metrics were reported
-    assertThat(registry.counters()).hasSize(1);
+    assertThat(registry.counters()).hasSize(3);
     Counter counter = registry.counters().findFirst().orElseThrow(AssertionError::new);
-    assertThat(counter.id().name()).isEqualTo("aws.request.requestCount");
-    assertThat(counter.id().tags()).contains(Tag.of("serviceName", "Amazon S3"));
+    assertThat(counter.id().name()).isEqualTo("ipc.client.call");
+    assertThat(counter.id().tags()).contains(Tag.of("http.status", "200"));
     assertThat(counter.actualCount()).isEqualTo(1);
-    assertThat(registry.gauges()).hasSize(3);
-
-    // Verify that the tag that comes from the request handler is present on an
-    // arbitrary gauge
-    Gauge gauge =
-        registry
-            .gauges()
-            .filter(Functions.nameEquals("aws.request.httpClientPoolAvailableCount"))
-            .findFirst()
-            .orElseThrow(AssertionError::new);
-    assertThat(gauge.id().tags())
-        .contains(
-            Tag.of(
-                SpectatorRequestMetricCollector.DEFAULT_HANDLER_CONTEXT_KEY.getName(),
-                account.getName()));
+    assertThat(registry.timers()).hasSize(3);
```

Comment:
Possible to assert on similar things here?  I'm curious what metrics we get, and what the tags look like.

Reply:
Timers and Counters are looking mostly alike. Counters have a couple of more pairs of tags. These tag key-values are coming from [this](https://github.com/Netflix/spectator/blob/v1.0.6/spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcMetric.java#L43-L62).

Here is a timer snapshot:
<img width="1246" height="783" alt="image" src="https://github.com/user-attachments/assets/6b6c9aed-ad36-44ff-af6d-7ff0f38717bc" />